packages/drivers/odsp-driver/src/odspDriverUrlResolverForShareLink.ts

```diff
@@ -213,13 +213,34 @@ export class OdspDriverUrlResolverForShareLink implements IUrlResolver {
 		const odspResolvedUrl = getOdspResolvedUrl(resolvedUrl);
 
 		const shareLink = await this.getShareLinkPromise(odspResolvedUrl);
-		const shareLinkUrl = new URL(shareLink);
 
-		let actualDataStorePath = dataStorePath;
+		return this.appendLocatorParams(shareLink, resolvedUrl, dataStorePath, packageInfoSource);
+	}
+
+	/**
+	 * Appends the store locator properties to the provided base URL. This function is useful for scenarios where an application
+	 * has a base URL (for example a sharing link) of the Fluid file, but does not have the locator information that would be used by Fluid
+	 * to load the file later.
+	 * @param baseUrl - The input URL on which the locator params will be appended.
+	 * @param resolvedUrl - odsp-driver's resolvedURL object.
+	 * @param dataStorePath - The relative data store path URL.
+	 * For requesting a driver URL, this value should always be '/'. If an empty string is passed, then dataStorePath
+	 * will be extracted from the resolved url if present.
+	 * @returns The provided base URL appended with odsp-specific locator information
+	 */
+	public async appendLocatorParams(
+		baseUrl: string,
+		resolvedUrl: IResolvedUrl,
+		dataStorePath: string,
+		packageInfoSource?: IContainerPackageInfo,
+	) {
+		const url = new URL(baseUrl);
+		const odspResolvedUrl = getOdspResolvedUrl(resolvedUrl);
+
 		// If the user has passed an empty dataStorePath, then extract it from the resolved url.
-		if (dataStorePath === "" && odspResolvedUrl.dataStorePath !== undefined) {
-			actualDataStorePath = odspResolvedUrl.dataStorePath;
-		}
+		const actualDataStorePath = dataStorePath
```

Comment:
I was going to ask whether this would work : 

actualDataStorePath = dataStorePath ?? odspResolvedUrl.dataStorePath ?? "";

But on the original if statement we had a special treatment for when dataStorePath === ""  (not undefined),
In this new one correct in case dataStorePath === "" we would get  "" - is that what we want ?

Reply:
`actualDataStorePath = dataStorePath ?? odspResolvedUrl.dataStorePath ?? "";` -- this will not work for us here, since if the dataStorePath is "", actualDataStorePath will still get assigned "" since it is a non-null value. (?? would only check for nullish values.)

`actualDataStorePath = "" ?? "a" ?? "b"` will result in `actualDataStorePath = ""`

So I updated the code to using `? :` operator:
`actualDataStorePath = "" ? "" : "b" || "";`   //=b
`actualDataStorePath = "a" ? "a" : "b" || "";`  //=a
`actualDataStorePath = "" ? "" : "" || "";`  //=""

I have tried to preserve the original logic, but let me know if you still think I haven't.